etna/packages/etna-js/upload/workers/uploader.jsx

```diff
@@ -174,14 +176,18 @@ export default (uploader) => {
 
     remove: (upload) => uploader.dispatch({ type: 'REMOVE_UPLOAD', upload }),
 
-    timeout: () => {
-      ++uploader.timeouts;
-      if (uploader.timeouts == uploader.maxTimeouts) {
-        uploader.reset();
+    timeout: (upload) => {
+      const { url } = upload;
```

Comment:
Does timeout here actually mean any kind of failure? I.e. a 422 lands here, but is not a "timeout"...

Reply:
I also was slightly confused by this terminology.  I think the original author's intent was for timeout to be like "gating multiple failed requests" to prevent server being inundated with requests that cannot succeed.  So yes, any error is the approach I went with since the idea I believe is to generally gracefully pause for any reason that the server continuously rejects requests.

Reply:
@graft What would you prefer the terminology and behavior be by the way?  I can see how the terminology is confusing, but I definitely think pausing downloads when _any_ general error repeats itself 5 times is worth having to prevent a bug from forcing retries constantly and hurting the server.